.cursor/rules/design.mdc

```diff
@@ -0,0 +1,204 @@
+
+## Component Conventions
+
+### File Structure:
+
+- Design system components: `apps/dashboard/shared/components/design-system/[ComponentName].tsx` or `apps/dashboard/shared/components/design-system/[category]/[ComponentName].tsx`
+- Feature-specific: `apps/dashboard/shared/components/[category]/[ComponentName].tsx`
+- UI primitives: `apps/dashboard/shared/components/ui/[component-name].tsx` (for internal use only)
+
+### Component Pattern:
+
+```typescript
+import * as React from "react";
+import { cva, type VariantProps } from "class-variance-authority";
+import { cn } from "@/shared/utils/";
+
+export const componentVariants = cva(
+  "base-classes-here",
+  {
+    variants: {
+      variant: {
+        default: "default-styles",
+        secondary: "secondary-styles",
+      },
+      size: {
+        sm: "small-styles",
+        md: "medium-styles",
+      },
+    },
+    defaultVariants: {
+      variant: "default",
+      size: "md",
+    },
+  },
+);
+
+export interface ComponentProps
+  extends React.HTMLAttributes<HTMLDivElement>,
+    VariantProps<typeof componentVariants> {
+  asChild?: boolean; // When polymorphism is needed
+}
+
+export const Component: React.FC<ComponentProps> = ({
+  className,
+  variant,
+  size,
+  ...props
+}) => {
+  return (
+    <div
+      className={cn(componentVariants({ variant, size }), className)}
+      {...props}
+    />
+  );
+};
+
+// Export both component and variants for flexibility
+export { Component, componentVariants };
+```
+
+### Required Practices:
+
+- Use `class-variance-authority` for variant management
+- Use `cn()` utility for class merging (from `@/shared/utils/`)
+- Support `className` prop override
+- Export both component and variants
+- Use semantic HTML elements with proper ARIA attributes
+- Add focus-visible styles, never remove focus outlines
+
+### Accessibility Requirements:
+
+- Buttons must have proper roles and keyboard support
+- Inputs must be labeled via `aria-label` or associated `<label>`
+- Interactive elements must have focus-visible styles
+- Use semantic HTML elements when possible
+
+## Testing & Documentation:
+
+- Consider adding Storybook stories for complex components
+- Test with screen readers and keyboard navigation
+- Ensure color contrast meets WCAG standards
+
+## Do / Don't
+
+✅ **Do:**
+
+- **Always prefer design-system components first**
+- Use design tokens from globals.css
+- Follow the established folder structure
+- Export components from appropriate index.ts files
+- Use class-variance-authority for variants
+- Compose from existing design-system components
+
+❌ **Don't:**
+
+- Use hex colors (`#RRGGBB`) - always use design tokens
+- Create inline styles (except rare documented cases)
+- Use UI primitives directly in features (use design-system components instead)
+- Duplicate components that already exist in `design-system/` folder
+- Skip accessibility considerations
+- Ignore the established import patterns
```

Comment:
This design system documentation is comprehensive and well-structured, providing clear guidance on component architecture, styling practices, and accessibility requirements. One potential improvement would be to explicitly emphasize error handling and edge case considerations within components. For instance, including guidance on how to gracefully handle missing or invalid props, asynchronous loading states, or fallback UI in components could enhance robustness. Additionally, explicitly recommending or integrating testing strategies for such edge cases in the 'Testing & Documentation' section would further strengthen component reliability.